docs/client-specifications.rst

```diff
@@ -0,0 +1,246 @@
+.. _client-specifications:
+
+Client Specifications
+=====================
+
+Foreword
+--------
+
+During years, the sole and unique client was Gecko (the platform behind Firefox, Thunderbird..). In order to reach out new platforms and products, a fully-featured Rust client was built, using the application-services components (Viaduct, NSS, ...).
+Unfortunately, this cross-platform was not leveraged by the experimentation team, which built another ad-hoc client. In front of this situation, we are obliged to admit that our idea of having a single client of reference is dead. Instead, we are going to provide specifications for Remote Settings clients, to at least mitigate the consequences of clients fragmentation.
+
+That being said, we still **strongly discourage** the implementation of ad-hoc clients.
+
+We distinguish two major use-cases:
+
+- authenticated write operations, ie. publish data;
+- anonymous read operations, ie. fetch data from within our products.
+
+Since the former does not take place on clients, it matters less than the latter, which has a major impact on traffic and our servers load.
+
+Existing Clients
+----------------
```

Comment:
Should we include some sort of (as of `<date>`) mention here? Just because a table like this can quickly become outdated